src/transformers/generation/utils.py

```diff
@@ -1474,6 +1495,8 @@ def generate(
             encoder_input_ids=inputs_tensor,
             prefix_allowed_tokens_fn=prefix_allowed_tokens_fn,
             logits_processor=logits_processor,
+            tokenizer=tokenizer,
+            device=inputs_tensor.device,
```

Comment:
I hope passing a device into the processor directly, works for the multi-gpu `generate`. 

Actually, this is quite handy to be able to init tensors in their devices, while init the processor. Especially when we make `compile` compatible processors, where we already moved to init some of arguments in tensor format.